backend/tests/test_fhir_service.py

```diff
@@ -120,7 +141,8 @@ def test_pre_validation_failed(self):
         self.assertEqual(error.exception.message, expected_msg)
         self.imms_repo.update_immunization.assert_not_called()
 
-    def test_post_validation_failed(self):
+    def test_post_validation_failed_get_by_all(self):
```

Comment:
This feels like two separate tests (which would make the mocking easier to understand)